torch/linalg/__init__.py

```diff
@@ -344,49 +381,78 @@
+
+For a complex Hermitian (or real symmetric) matrix :math:`A`, this is defined as
 
-For a single matrix :attr:`input`, the tensor of eigenvalues `w` and the tensor of eigenvectors
-`V` decompose the :attr:`input` such that `input = V diag(w) Vᴴ`, where `Vᴴ` is the transpose of `V`
-for real-valued :attr:`input`, or the conjugate transpose of `V` for complex-valued :attr:`input`.
+.. math::
+
+    A = V \operatorname{diag}(L) V^{\text{H}}
+
+where :math:`V^{\text{H}}` denotes the conjugate transpose in the complex case and the transpose in
+the real case. The matris :math:`V` (and thus :math:`V^{\text{H}}`)
+is orthogonal in the real case, and unitary in the complex case.
+
+The decomposition is represented as a namedtuple `(eigenvalues, eigenvectors)`, which
+corresepond to `(L, V)` above. If :attr:`input` is a batch of matrices, then `L`, `V` are also
+batched with the same batch dimensions.
+
+The eigenvalues are returned in ascending order.
 
-Since the matrix or matrices in :attr:`input` are assumed to be Hermitian, the imaginary part of their diagonals
-is always treated as zero. When :attr:`UPLO` is "L", its default value, only the lower triangular part of each
-matrix is used in the computation. When :attr:`UPLO` is "U" only the upper triangular part of each matrix is used.
+:attr:`input` is assumed to be Hermitian (resp. symmetric), but this is not checked internally.
+When :attr:`UPLO` is `'L'` (default), only the lower triangular part of each matrix is used in the computation.
+When :attr:`UPLO` is `'U'`, only the upper triangular part of each matrix is used.
 
-Supports input of float, double, cfloat and cdouble dtypes.
+Supports :attr:`input` of float, double, cfloat and cdouble dtypes.
+The eigenvalues will always be real-valued, even if :attr:`input` is complex.
+The eigenvectors will have the same dtype as :attr:`input`.
 
-.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+.. note:: When :attr:`input` is on a CUDA device, this function synchronizes that device with the CPU.
 
-.. note:: The eigenvalues/eigenvectors are computed using LAPACK's `syevd` and `heevd` routines for CPU inputs,
-          and MAGMA's `syevd` and `heevd` routines for CUDA inputs.
+.. note:: This function is computed using LAPACK's and MAGMA's `syevd` and `heevd` routines for
+          CPU and CUDA symmetric and Hermitian inputs respectively.
 
 .. note:: The eigenvalues of real symmetric or complex Hermitian matrices are always real.
 
-.. note:: The eigenvectors of matrices are not unique, so any eigenvector multiplied by a constant remains
-          a valid eigenvector. This function may compute different eigenvector representations on
-          different device types. Usually the difference is only in the sign of the eigenvector.
+.. warning:: The eigenvectors of a matrix are not unique. Any eigenvector may be multiplied by `-1`
```

Comment:
The second sentence here isn't actually saying anything because it just says that vectors can be multiplied by certain scalars.

Reply:
Fixed